### PR TITLE
readthedocs: Add contribution guidelines

### PR DESCRIPTION
**Pull request by sunithabeeram**
Added guidelines for creating issues, changes, PRs.

### CODE REVIEW

File: docs/contribution_guidelines.rst [hunk 1]
```diff
@@ -54,13 +126,79 @@ All source code files should have license headers. To automatically add the head
 
 .. note::
 
-If you checkin third-party code or files, do not add license headers for them. Follow these instructions to ensure we
-are compliant with Apache Licensing process. <TBD>
+If you checkin third-party code or files, ensure they are distributed under Apache, MIT or BSD-3 licenses.
```
Comment (snleee) [inline docs/contribution_guidelines.rst]: Let's put the link for the below page. It gives us the extensive list of allowed/not-allowed licenses
https://www.apache.org/legal/resolved.html#what-can-we-include-in-an-asf-project-category-a
https://www.apache.org/legal/resolved.html#what-can-we-maybe-include-in-an-asf-project-category-b
https://www.apache.org/legal/resolved.html#what-can-we-not-include-in-an-asf-project-category-x

File: docs/contribution_guidelines.rst [hunk 2]
```diff
@@ -54,13 +126,79 @@ All source code files should have license headers. To automatically add the head
...
 * Addressing comments and Rebasing
+Once you receive comments on github on your changes, be sure to respond to them on github and address the concerns.
+If any discussions happen offline for the changes in question, make sure to capture the outcome of the discussion, so others can follow along as well.
+
+It is possible that while your change is being reviewed, other changes were made to the master branch. Be sure to pull rebase your change on the new changes thus:
+
+.. code-block:: none
+
+  # commit your changes
+  $git add <updated files>
```
Comment (snleee) [inline docs/contribution_guidelines.rst]: space after `$`

File: docs/contribution_guidelines.rst [hunk 3]
```diff
@@ -54,13 +126,79 @@ All source code files should have license headers. To automatically add the head
...
+  $git pull
+  $git checkout <your issue branch>
+  $git rebase master
+
+At this time, if rebase flags any conflicts, resolve the conflicts and follow the instructions provided by the rebase command.
+
+Run additional tests/validations for the new changes and update the PR by pushing your changes:
+
+.. code-block:: none
+  $git push origin <your issue branch>
```
Comment (snleee) [inline docs/contribution_guidelines.rst]: same here

File: docs/contribution_guidelines.rst (hunk 1 above)
Comment (sunithabeeram) [reply docs/contribution_guidelines.rst]: Added

File: docs/contribution_guidelines.rst (hunk 2 above)
Comment (sunithabeeram) [reply docs/contribution_guidelines.rst]: Fixed

**Pull request merged**